### PR TITLE
Initial Commit with CLI parsing

### PR DESCRIPTION
**Pull request by sridamul**
In this PR,
- I set up the project
- Implemented the CLI module functions (not all)
- Unit test coverage for CLI module classes
- Integrated spotbugs
- Used maven assembly plugin to build jar with dependencies

cc: @berviantoleo, @iamrajiv, @jonesbusy, @gounthar

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: gh-integration/pom.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.jenkins.plugin-modernizer</groupId>
+        <artifactId>plugin-modernizer-parent-pom</artifactId>
+        <version>${revision}${changelist}</version>
+        <relativePath>../pom.xml</relativePath>
```
Comment (jonesbusy) [inline gh-integration/pom.xml:10]: If i'm not mistaken the relativePath is optional if found of direct parent directory

File: core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.jenkins.plugin-modernizer</groupId>
+        <artifactId>plugin-modernizer-parent-pom</artifactId>
+        <version>${revision}${changelist}</version>
```
Comment (jonesbusy) [inline core/pom.xml:9]: If we are using CD also for the tool (Which I like very much) I think we are missing some files like https://www.jenkins.io/doc/developer/publishing/releasing-cd/#update-maven-pom-and-config

Also incrementals must be enabled (I would suggest to check some plugin with CD enabled)

Comment (sridamul) [reply core/pom.xml:9]: Will take a look into it

File: core/pom.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>core</artifactId>
```
Comment (jonesbusy) [inline core/pom.xml:13]: I think using a suffix is more clear for the artifact ID of child module

`plugin-modernizer-parent-core`
`plugin-modernizer-parent-cli`

Comment (sridamul) [reply core/pom.xml:13]: is `plugin-modernizer-core, plugin-modernizer-cli, plugin-modernizer-recipes, plugin-modernizer-integration` ok? (or) `parent` should be added to each module?

Comment (jonesbusy) [reply core/pom.xml:13]: Looks good.  `parent` (or `pom`) is only for the top level pom module

File: core/pom.xml (hunk 2 above)
Comment (sridamul) [reply core/pom.xml]: Hey! @jonesbusy I have enabled incrementals and added Dependabot. Will configure others later.

**Review by jonesbusy — APPROVED** [review]

**Pull request merged**